### PR TITLE
PAE-103 - Changes the text input  to a dropdown when the studio_home.enable_global_staff_optimization is activated

### PR DESCRIPTION
**Pull request by elvinquero**
Right now there is a Waffle Switch that when activated, it displays a text input to filter courses by org in the Studio Home page. The default behavior when no org is selected is to display the page without a single course.

This PR changes:
1) The text input is changed to a Dropdown. The **ORGANIZATION_APP** has to be activated in the CMS for it to recover all the orgs from discovery.

Guide to activate the dropdown:
1. Make sure the **ORGANIZATIONS_APP** is set to True on the FEATURES dict on edx-platform/cms/envs/common.py https://github.com/Pearson-Advance/edx-platform/blob/3792f3d740d76b245f7ff05492ede2ce9a637129/cms/envs/common.py#L283
2. Create and activate the **studio_home.enable_global_staff_optimization** Waffle Switch.

On the Stage and Production server, the organizations app will be activated through ANSIBLE variables, so there is no need to activated on the PR.

### CODE REVIEW

File: cms/envs/devstack.py [hunk 1]
```diff
@@ -209,6 +209,10 @@ def should_show_debug_toolbar(request):
 # Dummy secret key for dev
 SECRET_KEY = '85920908f28904ed733fe576320db18cabd7b6cd'
 
+FEATURES.update({
```
Comment (Jacatove) [inline cms/envs/devstack.py:212]: @elvinquero I would put this here: https://github.com/Pearson-Advance/edx-platform/blob/3792f3d740d76b245f7ff05492ede2ce9a637129/cms/envs/common.py#L164
and keep in mind that you will need to set this variable in ansible in order to retrieve the variable in production, take a look at this: https://github.com/Pearson-Advance/edx-platform/blob/3792f3d740d76b245f7ff05492ede2ce9a637129/cms/envs/production.py#L273

File: cms/djangoapps/contentstore/views/course.py [hunk 2]
```diff
@@ -505,7 +505,11 @@ def course_listing(request):
     optimization_enabled = GlobalStaff().has_user(request.user) and \
         WaffleSwitchNamespace(name=WAFFLE_NAMESPACE).is_enabled(u'enable_global_staff_optimization')
 
-    org = request.GET.get('org', '') if optimization_enabled else None
+    organizations = get_organizations() if optimization_enabled else []
+    org_names_list = [(org["short_name"]) for org in organizations]
+
+    selected_org = request.GET.get('org', '')
+    org = selected_org if optimization_enabled and selected_org != '' else None
```
Comment (Jacatove) [inline cms/djangoapps/contentstore/views/course.py:512]: @elvinquero  What do you think about this?
```suggestion
    selected_org = request.GET.get('org', '')
    org = selected_org if optimization_enabled and selected_org else None
```
Is there a reason to validate that selected_org isn't empty?

File: cms/djangoapps/contentstore/views/course.py [hunk 3]
```diff
@@ -562,7 +566,8 @@ def format_library_for_view(library):
         u'rerun_creator_status': GlobalStaff().has_user(user),
         u'allow_unicode_course_id': settings.FEATURES.get(u'ALLOW_UNICODE_COURSE_ID', False),
         u'allow_course_reruns': settings.FEATURES.get(u'ALLOW_COURSE_RERUNS', True),
-        u'optimization_enabled': optimization_enabled
+        u'optimization_enabled': optimization_enabled,
+        u'org_names_list': org_names_list
```
Comment (Jacatove) [inline cms/djangoapps/contentstore/views/course.py:570]: @elvinquero Add a trailing comma.
```suggestion
        u'org_names_list': org_names_list,
```

**Review by Jacatove — CHANGES_REQUESTED** [review]

**Review by diegomillan — APPROVED** [review]

**Review by Jacatove — CHANGES_REQUESTED** [review]

**Comment by elvinquero** [comment]
On a local devstack, there is more than just one discovery. Localhost, discovery.main.localhost. The Studio app will bring the Orgs based on what Discovery is connected to. I believe that on stage and prod settings, just one discovery is used, so it won't cause a problem.

**Comment by elvinquero** [comment]
As the ANSIBLE variables will override the default settings. There is no need to activate the ORGANIZATIONS_APP on this PR, however in order to test it, people will need set it to `True` on https://github.com/Pearson-Advance/edx-platform/blob/3792f3d740d76b245f7ff05492ede2ce9a637129/cms/envs/common.py#L283

File: cms/envs/common.py [hunk 4]
```diff
@@ -280,7 +280,7 @@
     # Special Exams, aka Timed and Proctored Exams
     'ENABLE_SPECIAL_EXAMS': False,
 
-    'ORGANIZATIONS_APP': False,
+    'ORGANIZATIONS_APP': True,
```
[the remark this answers is not in the document]
Comment (elvinquero) [reply cms/envs/common.py]: Done! And I updated the PR description to let others know how to activate the filter feature.

**Review by diegomillan — APPROVED** [review]
Understood @elvinquero, ansible will handle the activation. Reapproved.

**Review by Jacatove — APPROVED** [review]

File: cms/djangoapps/contentstore/views/course.py [hunk 5]
```diff
@@ -505,7 +505,11 @@ def course_listing(request):
     optimization_enabled = GlobalStaff().has_user(request.user) and \
         WaffleSwitchNamespace(name=WAFFLE_NAMESPACE).is_enabled(u'enable_global_staff_optimization')
 
-    org = request.GET.get('org', '') if optimization_enabled else None
+    organizations = get_organizations() if optimization_enabled else []
+    org_names_list = [(org["short_name"]) for org in organizations]
```
Comment (Squirrel18) [inline cms/djangoapps/contentstore/views/course.py:509]: Add a safe getter for "short_name". Also, do not mix single and double quotes. @elvinquero 
nit: you can remove this variable as it's only used once.

File: cms/djangoapps/contentstore/views/course.py [hunk 6]
```diff
@@ -505,7 +505,11 @@ def course_listing(request):
...
+
```
Comment (Squirrel18) [inline cms/djangoapps/contentstore/views/course.py:510]: Unnecessary blank line. @elvinquero

**Review by Squirrel18 — CHANGES_REQUESTED** [review]

**Comment by elvinquero** [comment]
@Squirrel18 suggestions were applied.

**Review by Squirrel18 — CHANGES_REQUESTED** [review]

**Pull request merged**